### PR TITLE
SMP/E workflow -- merge together with PR 1008

### LINKED ISSUE #881: z/OSMF SMP/E Workflow
Add z/OSMF SMP/E Workflow to the ZOWE pax file.

### PR DESCRIPTION
**Pull request by abastyr**
Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type

- [ ] Bugfix 
- [X] Feature 
- [ ] Other... Please describe:

#### Relevant issues

Fixes #881

#### Changes proposed in this PR

- Adds VTL templates
- Adds the template for the z/OSMF workflow
- Adds the generation of the SMP/E workflow from the workflow template

#### Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

#### Does this PR do something the person installing Zowe should know about?

Adds the SMP/E Workflow to the F4 data set.

---
* Affected function: _general area of interest_ *

---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 

Doc issue/PR number:

### CODE REVIEW

File: smpe/pax/ZOSMF/smpe_workflow.xml [hunk 1]
```diff
@@ -309,6 +309,7 @@
 		<variableValue name="fsclass" noPromptIfSet="false" required="false" scope="instance"/>
 		<variableValue name="fdclass" noPromptIfSet="false" required="false" scope="instance"/>
 		<variableValue name="fmclass" noPromptIfSet="false" required="false" scope="instance"/>
+		<variableValue name="ibmTemplate" noPromptIfSet="false" required="false" scope="instance"/>
```
Comment (abastyr) [inline smpe/pax/ZOSMF/smpe_workflow.xml]: This variable cannot be here. It is VTL templates variable only, in the workflow we expect the default to be used so the value cannot be prompted...

File: smpe/pax/ZOSMF/vtls/ZWE5MKD.vtl [hunk 2]
```diff
@@ -83,15 +96,43 @@
...
+#if($ibmTemplate == 'NO')
+  ${pathPrefix}usr/lpp/zowe
+#end
+#if($ibmTemplate == 'YES')
   -PathPrefix-usr/lpp/zowe
+#end
 $$
 //DIRS     DD *
   # do NOT change these definitions
   SMPE
```
Comment (OnnoVdT) [inline smpe/pax/ZOSMF/vtls/ZWE5MKD.vtl]: add new line after SMPE to create "workflow" directory

File: smpe/bld/SMPMCS.txt [hunk 3]
```diff
@@ -62,6 +62,12 @@
 ++HFS(ZWEPAX07)      SYSLIB(SZWEZFS ) DISTLIB(AZWEZFS ) RELFILE(4)
    SHSCRIPT(ZWESHPAX,PRE,POST)
    BINARY            PARM(PATHMODE(0,7,5,5)) .
+++HFS(ZWEWRF01)      SYSLIB(SZWEZFS ) DISTLIB(AZWEZFS ) RELFILE(4)
+   SHSCRIPT(ZWESHPAX,PRE,POST)
+   BINARY            PARM(PATHMODE(0,7,5,5)) .
```
[the remark this answers is not in the document]
Comment (uzust01) [reply smpe/bld/SMPMCS.txt]: I tested the workflow with text and I think there was some problem. Binary works, but I can retest.

File: smpe/bld/SMPMCS.txt [hunk 4]
```diff
@@ -62,6 +62,12 @@
...
+++HFS(ZWE0YML)      SYSLIB(SZWEZFS ) DISTLIB(AZWEZFS ) RELFILE(4)
```
[the remark this answers is not in the document]
Comment (uzust01) [reply smpe/bld/SMPMCS.txt]: Renamed it to ZWEYML01.

File: smpe/bld/SMPMCS.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (OnnoVdT) [reply smpe/bld/SMPMCS.txt]: OK, we'll go with binary and adjust if needed

**Comment by OnnoVdT** [comment]
The awk script is not working on z/OS, so I rewrote its function. I used REXX to avoid issues with escaping special characters. I also moved the zosmf build stuff out of the pax directory, as this should only hold data to be merged into the build output. 
My test build is successful, see https://wash.zowe.org:8443/job/zowe-install-packaging/job/smpe_workflow-onno/6/consoleFull

My updates in PR https://github.com/zowe/zowe-install-packaging/pull/1008 must be merged the same time this PR is merged.

**Comment by OnnoVdT** [comment]
closing this request, changes went in via PR https://github.com/zowe/zowe-install-packaging/pull/1008

**Pull request closed without merging**